packages/ses/src/scope-handler.js

```diff
@@ -67,7 +67,7 @@ export function createScopeHandler(
         // fall through
       }
 
-      // Properties of the global.
+      // Properties of the endowments.
       if (prop in endowments) {
         // Use reflect to defeat accessors that could be
         // present on the endowments object itself as `this`.
```

Comment:
Now that I'm looking at it, I can answer the questions from the keybase chat.

This, and the `reflectSet`s below, address the vulnerability I was remembering. By using the `reflect*` forms, we feed in the safe `globalObject` as `this`, whereas just reading or setting the property on `endowments` (now `localObject`?) would pass `endowments` as `this` to those accessors. This would enable navigational access to the other global lexicals, violating our "access only by statically naming them" invariant.

Your question though is whether we should change from `globalObject` to `undefined`. Since we agree that accessors here are *only* to support live bindings within the shim, and will be absent from the proposal and from non-shim implementations, we need only ask what the live bindings need. They don't need a `this`. POLA therefore suggests that `undefined` is a better choice.

Reply:
Since we agree that changing the reflect calls `globalObject` to `undefined` is a good idea, feel free to do it in this PR. If you don't, please add a TODO or file an issue.